src/Parsers/Kusto/ParserKQLQuery.cpp

```diff
@@ -373,9 +373,9 @@ bool ParserKQLQuery::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
 
     uint16_t bracket_count = 0;
 
-    while ((pos->type < TokenType::EndOfStream ||
```

Comment:
Ok, but it works exactly the same.

Reply:
before we use  pos->isEnd() to validate, but if  when reach max_query_size,  the iterator already passed the end

Reply:
The code looks the same: `bool isValid() { return get().type < TokenType::EndOfStream; }`

Reply:
oh, yes, changed according to @antaljanosbenjamin 's suggestion

Reply:
Because `pos->isEnd()` would be `false` in case of `TokenType::ErrorMaxQuerySizeExceeded`, thus `!pos->isEnd()` wouldn't terminate the loop. Meanwhile `pos->isValid()` will be false in case of `TokenType::ErrorMaxQuerySizeExceeded`, thus stopping the loop. Do I miss something?